pyanaconda/modules/storage/partitioning/automatic.py

```diff
@@ -0,0 +1,268 @@
+    def type(self):
+        """Type of a filesystem used on the partitions."""
+        return self._type
+
+    def set_type(self, scheme):
+        """Set the partitioning scheme.
+
+        :param scheme: an instance of AutoPartitioningType
+        """
+        self._type = scheme
+        self.type_changed.emit()
+        log.debug("Type is set to '%s'.", scheme)
+
+    @property
+    def fstype(self):
+        """Type of a filesystem used on the partitions."""
+        return self._fstype
+
+    def set_fstype(self, fstype):
+        """Set the type of a filesystem used on the partitions.
+
+        :param fstype: a string with the filesystem type
+        """
+        self._fstype = fstype
+        self.fstype_changed.emit()
+        log.debug("Filesystem type is set to '%s'.", fstype)
+
+    @property
+    def nohome(self):
+        """Do not create a /home partition."""
+        return self._nohome
+
+    def set_nohome(self, nohome):
+        """Enable or disable creation of a /home partition.
+
+        :param nohome: a boolean value
+        """
+        self._nohome = nohome
+        self.nohome_changed.emit()
+        log.debug("Nohome is set to '%s'.", nohome)
+
+    @property
+    def noboot(self):
+        """Do not create a /boot partition."""
+        return self._noboot
+
+    def set_noboot(self, noboot):
+        """Enable or disable creation of a /boot partition.
+
+        :param noboot: a boolean value
+        """
+        self._noboot = noboot
+        self.noboot_changed.emit()
+        log.debug("Noboot is set to '%s'.", noboot)
+
+    @property
+    def noswap(self):
+        """Do not create a swap partition."""
+        return self._noswap
+
+    def set_noswap(self, noswap):
+        """Enable or disable creation of a swap partition.
+
+        :param noswap: a boolean value
+        """
+        self._noswap = noswap
+        self.noswap_changed.emit()
+        log.debug("Noswap is set to '%s'.", noswap)
+
+    @property
+    def encrypted(self):
+        """Should all devices with support be encrypted by default?"""
+        return self._encrypted
+
+    def set_encrypted(self, encrypted):
+        """Set if all devices with support should be encrypted by default.
+
+        :param encrypted: a boolean value
+        """
+        self._encrypted = encrypted
+        self.encrypted_changed.emit()
+        log.debug("Encrypted is set to '%s'.", encrypted)
+
+    @property
+    def cipher(self):
+        """Encryption algorithm used to encrypt the filesystem."""
+        return self._cipher
+
+    def set_cipher(self, cipher):
+        """Set the encryption algorithm used to encrypt the filesystem.
+
+        :param cipher: a name of an algorithm
+        """
+        self._cipher = cipher
+        self.cipher_changed.emit()
+        log.debug("Cipher is set to '%s'.", cipher)
+
+    @property
+    def passphrase(self):
+        """Default passphrase for all encrypted devices."""
+        return self._passphrase
```

Comment:
This just feels weird to me. I know we need it now, but we shouldn't provide password back from module. I don't think we will really need that when this spoke is finished. 

So could you please add TODO here to remove getter when module will be completed please?

Reply:
Yeah, something similar to the note in the user module could be used:
https://github.com/rhinstaller/anaconda/blob/master/pyanaconda/modules/user/user_interface.py#L66

One more reason to concentrate on moving the logic on these modules as soon as possible. :)

Reply:
Fixed in the DBus interface.